### ISSUE TITLE
Implement generic MPLS module

### ISSUE THREAD

**Issue by ipspace** [ipspace/netsim-tools#60]
We need support for other MPLS label allocation mechanisms beyond SR-MPLS. The initial implementation should support LDP and BGP-LU.

Configurable parameters:

**mpls.protocol**: a string or a list of protocols _ldp_, _bgp_, _sr_. Default: _ldp_
**mpls.advertise**: a string or a list of link roles. Default: _loopback_
**mpls._protocol_.enable**: a list of link roles or _internal_/_external_ for BGP. Default: enable on all applicable links/sessions.

**Comment by ssasso** [comment]
Maybe it could be worth to add two "levels" of parameters/flags to enable it, at node and also at link level?

```
nodes:
  pe1:
    mpls.enable: True
  ce1:
    mpls.enable: False

links:
- pe1:
  ce1:
  mpls.enable: False
```

And, at the Ansible template level,
if **node.mpls.enable** the "global config" for MPLS shall be performed, and then MPLS is enabled on the different links only if **node.mpls.enable && link.mpls.enable**.

**Comment by ipspace** [comment]
That's exactly the direction we've been taking with OSPF, IS-IS and BFD. To disable BFD on a link we use **bfd: false**. Likewise, we could use **mpls: false**.

In the final node.interfaces list, **mpls** attribute indicates we need to do something on the interface. For example, **mpls.protocol** equal to **ldp** would turn into **mpls ip** on Cisco IOS.

**Comment by ipspace** [comment]
Implemented in #226
